src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -654,15 +654,27 @@ public long getTaskCount(SearchTaskRequest searchTaskRequest) {
      */
     @Transactional
     public void terminateTask(String taskId, TerminateInfo terminateInfo) {
-        //Find and Lock Task
-        TaskResource task = findByIdAndObtainLock(taskId);
-        //Update cft task and terminate reason
-        task.setState(CFTTaskState.TERMINATED);
-        task.setTerminationReason(terminateInfo.getTerminateReason());
-        //Perform Camunda updates
-        camundaService.deleteCftTaskState(taskId);
-        //Commit transaction
-        cftTaskDatabaseService.saveTask(task);
+        TaskResource task = null;
+        try {
+            //Find and Lock Task
+            task = findByIdAndObtainLock(taskId);
```

Comment:
So does the transaction we open here close when we return?

Reply:
Yes, and as it is read only call, doesn't make any difference either we rollback or commit